src/shared/sam/cli/samCliConfiguration.ts

```diff
@@ -3,85 +3,151 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import * as vscode from 'vscode'
-import { SettingsConfiguration } from '../../settingsConfiguration'
+import { getLogger } from '../../logger'
+import { fromPackage, migrateSetting, SettingsConfiguration } from '../../settingsConfiguration'
+import { stripUndefined, toRecord } from '../../utilities/collectionUtils'
+import { once } from '../../utilities/functionUtils'
+import { ClassToInterfaceType, keys } from '../../utilities/tsUtils'
+import { DefaultSamCliLocationProvider, SamCliLocationProvider } from './samCliLocator'
 
-export interface SamCliConfiguration {
-    /** Gets the current SAM CLI location from the VSCode settings store. */
-    getSamCliLocation(): string | undefined
+// TODO(sijaden): remove after a few releases
+async function migrateLegacySettings() {
```

Comment:
Moves all the SAM CLI settings to be under `aws.samcli` (would `aws.sam` be better?)
Perhaps some of the settings don't necessarily belong there, but this is how they're currently being used.